components/api_server/src/database/measurements.py

```diff
@@ -81,21 +62,15 @@ def recent_measurements(
     max_iso_timestamp: str = "",
     days: int = 7,
 ) -> dict[MetricId, list[Measurement]]:
-    """Return all recent measurements, or only those of the specified metrics."""
+    """Return all recent measurements."""
     max_iso_timestamp = max_iso_timestamp or iso_timestamp()
     min_iso_timestamp = (datetime.fromisoformat(max_iso_timestamp) - timedelta(days=days)).isoformat()
-    measurement_filter: dict[str, Any] = {"end": {"$gte": min_iso_timestamp}, "start": {"$lte": max_iso_timestamp}}
-    measurement_filter["metric_uuid"] = {"$in": list(metrics_dict.keys())}
-    projection = {"_id": False, "sources.entities": False, "entity_user_data": False}
-    measurements = database.measurements.find(
-        measurement_filter,
-        sort=[("start", pymongo.ASCENDING)],
-        projection=projection,
-    )
+    measurement_filter: dict[str, Any] = {"start": {"$lte": max_iso_timestamp}, "end": {"$gte": min_iso_timestamp}}
+    measurements = database.measurements.find(measurement_filter, sort=START_ASCENDING, projection=NO_SOURCE_DETAILS)
     measurements_by_metric_uuid: dict[MetricId, list] = {}
     for measurement_dict in measurements:
         metric_uuid = measurement_dict["metric_uuid"]
-        metric = metrics_dict[metric_uuid]
-        measurement = Measurement(metric, measurement_dict)
-        measurements_by_metric_uuid.setdefault(metric_uuid, []).append(measurement)
+        if metric := metrics_dict.get(metric_uuid):  # pragma: no feature-test-cover
```

Comment:
All calling code can handle getting an empty result `if not metric`?

Reply:
The `if` is needed because there may be measurements for metrics that have been deleted. I'll add a comment.